lerobot/common/robot_devices/cameras/opencv.py

```diff
@@ -251,45 +284,45 @@ def connect(self):
-        else:
-            self.camera = cv2.VideoCapture(self.camera_index)
-
-        if self.fps is not None:
-            self.camera.set(cv2.CAP_PROP_FPS, self.fps)
-        if self.width is not None:
-            self.camera.set(cv2.CAP_PROP_FRAME_WIDTH, self.width)
-        if self.height is not None:
-            self.camera.set(cv2.CAP_PROP_FRAME_HEIGHT, self.height)
-
-        actual_fps = self.camera.get(cv2.CAP_PROP_FPS)
-        actual_width = self.camera.get(cv2.CAP_PROP_FRAME_WIDTH)
-        actual_height = self.camera.get(cv2.CAP_PROP_FRAME_HEIGHT)
-
-        if self.fps is not None and not math.isclose(self.fps, actual_fps, rel_tol=1e-3):
-            raise OSError(
-                f"Can't set {self.fps=} for OpenCVCamera({self.camera_index}). Actual value is {actual_fps}."
-            )
-        if self.width is not None and self.width != actual_width:
-            raise OSError(
-                f"Can't set {self.width=} for OpenCVCamera({self.camera_index}). Actual value is {actual_width}."
-            )
-        if self.height is not None and self.height != actual_height:
-            raise OSError(
-                f"Can't set {self.height=} for OpenCVCamera({self.camera_index}). Actual value is {actual_height}."
-            )
+        with self.lock:
+            self.camera = VideoCapture(camera_idx)
+
+            if self.fps is not None:
+                self.camera.set(CAP_PROP_FPS, self.fps)
+            if self.width is not None:
+                self.camera.set(CAP_PROP_FRAME_WIDTH, self.width)
+            if self.height is not None:
+                self.camera.set(CAP_PROP_FRAME_HEIGHT, self.height)
+
+            actual_fps = self.camera.get(CAP_PROP_FPS)
+            actual_width = self.camera.get(CAP_PROP_FRAME_WIDTH)
+            actual_height = self.camera.get(CAP_PROP_FRAME_HEIGHT)
+
+            # Using `math.isclose` since actual fps can be a float (e.g. 29.9 instead of 30)
+            if self.fps is not None and not math.isclose(self.fps, actual_fps, rel_tol=1e-3):
+                # Using `OSError` since it's a broad that encompasses issues related to device communication
+                raise OSError(
+                    f"Can't set {self.fps=} for OpenCVCamera({self.camera_index}). Actual value is {actual_fps}."
+                )
+            if self.width is not None and self.width != actual_width:
+                raise OSError(
+                    f"Can't set {self.width=} for OpenCVCamera({self.camera_index}). Actual value is {actual_width}."
+                )
+            if self.height is not None and self.height != actual_height:
+                raise OSError(
+                    f"Can't set {self.height=} for OpenCVCamera({self.camera_index}). Actual value is {actual_height}."
+                )
 
-        self.fps = actual_fps
-        self.width = actual_width
-        self.height = actual_height
+            self.fps = int(actual_fps)
```

Comment:
Nit: For these lines do we want to use int() or round()? i.e. do we want to truncate these values or round them?
```
            self.fps = int(actual_fps)
            self.width = int(actual_width)
            self.height = int(actual_height)
```